go-update/src/main/java/org/reactome/release/goupdate/GoTermsUpdater.java

```diff
@@ -319,236 +255,272 @@ else if (goTermsFromFile.get(goID).containsKey(GoUpdateConstants.IS_OBSOLETE) &&
 		mainOutput.append(mismatchCount + " existing GO term instances in the database had mismatched categories when compared to the file (and were deleted from the database).\n");
 		mainOutput.append(obsoleteCount + " were obsolete. "+deletedCount+ " were actually deleted, and "+undeleteble.size()+" could not be deleted due to existing referrers.\n");
 		mainOutput.append(pendingObsoleteCount + " are pending obsolescence (and will probably be deleted at a future date).\n");
-		
-		reconcile(goTermsFromFile, goToECNumbers);
+		GoTermsReconciler reconciler = new GoTermsReconciler(this.adaptor);
+		reconciler.reconcile(goTermsFromFile, goToECNumbers);
 		
 		return mainOutput;
 	}
 
-	protected static Map<GKSchemaAttribute, Integer> getReferrerCounts(GKInstance inst) throws Exception
+	/**
+	 * Updates the relationships of GO terms.
+	 * @param goTermsFromFile - the GO terms from the GO file.
+	 * @param allGoInstances - a map of ALL GO instances from the database.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 * @throws InvalidAttributeValueException
+	 */
+	private void updateRelationships(Map<String, Map<String, Object>> goTermsFromFile, Map<String, List<GKInstance>> allGoInstances) throws InvalidAttributeException, Exception, InvalidAttributeValueException
 	{
-		Map<GKSchemaAttribute, Integer> referrersCount = new HashMap<>();
-		for (GKSchemaAttribute attrib : (Collection<GKSchemaAttribute>)inst.getSchemClass().getReferers())
+		for (String goId : goTermsFromFile.keySet())
 		{
-			@SuppressWarnings("unchecked")
-			Collection<GKInstance> referrers = inst.getReferers(attrib);
-			if ( referrers!=null &&  referrers.size() > 0)
+			List<GKInstance> goInsts = allGoInstances.get(goId);
+			Map<String, Object> goProps = goTermsFromFile.get(goId);
+			if (goInsts != null && !goInsts.isEmpty() && goProps != null && !goProps.isEmpty())
 			{
-				referrersCount.put(attrib, referrers.size());
+				for (GKInstance goInst : goInsts)
+				{
+					GoTermInstanceModifier goModifier = new GoTermInstanceModifier(this.adaptor, goInst, this.instanceEdit);
+					
+					if (goInst.getSchemClass().isa(ReactomeJavaConstants.GO_CellularComponent))
+					{
+						goModifier.updateRelationship(allGoInstances, goProps, GoUpdateConstants.IS_A, ReactomeJavaConstants.instanceOf);
+						goModifier.updateRelationship(allGoInstances, goProps, GoUpdateConstants.HAS_PART, "hasPart");
+						goModifier.updateRelationship(allGoInstances, goProps, GoUpdateConstants.PART_OF, ReactomeJavaConstants.componentOf);
+					}
+					// Update the instanace's "modififed".
+					goInst.getAttributeValuesList(ReactomeJavaConstants.modified);
```

Comment:
I don't think this `getAttributeValuesList` is needed

Reply:
I'm not always 100% sure with the Reactome Java API. In this case, I want to append to the existing values that are already in the database. I recall running into trouble once where I *didn't* retrieve the list prior to appending and then somehow wiped out the list. So that's probably why I did it this way.

Reply:
Ah true, I was wondering if that was the purpose of it. I don't know enough to prove or disprove this